web-components/src/components/timepicker/TimePicker.ts

```diff
@@ -0,0 +1,305 @@
+import { customElement, html, LitElement, property, internalProperty, PropertyValues } from "lit-element";
+import reset from "@/wc_scss/reset.scss";
+import styles from "./scss/module.scss";
+import "@/components/input/Input";
+import { ValidationRegex } from "@/utils/validations.ts";
+import { Input } from "@/components/input/Input";
+import { TIME_UNIT, Key } from "@/constants";
+import { ifDefined } from "lit-html/directives/if-defined";
+import { nothing } from "lit-html";
+
+export const timeUnits = [
+  TIME_UNIT.HOUR,
+  TIME_UNIT.MINUTE,
+  TIME_UNIT.SECOND,
+  TIME_UNIT.AM_PM,
+] as const;
+
+export const timeSpecificity = [
+  TIME_UNIT.HOUR,
+  TIME_UNIT.MINUTE,
+  TIME_UNIT.SECOND
+]
+
+const timePlaceholders = {
+  [TIME_UNIT.HOUR]: "HH",
+  [TIME_UNIT.MINUTE]: "MM",
+  [TIME_UNIT.SECOND]: "SS",
+  [TIME_UNIT.AM_PM]: "AM"
+}
+
+const timeUnitProps = (isTwentyFourHour: boolean) => {
+  return {
+    [TIME_UNIT.HOUR]: {
+      type: "number" as Input.Type,
+      min: 1,
+      max: isTwentyFourHour ? 23 : 12
+    },
+    [TIME_UNIT.MINUTE]: {
+      type: "number" as Input.Type,
+      min: 0,
+      max: 59
+    },
+    [TIME_UNIT.SECOND]: {
+      type: "number" as Input.Type,
+      min: 0,
+      max: 59
+    },
+    [TIME_UNIT.AM_PM]: {
+      type: "text" as Input.Type,
+      min: undefined,
+      max: undefined
+    }
+  }
+}
+
+export namespace TimePicker {
+  export type TimeUnit = typeof timeUnits[number];
+  export type TimeSpecificity = typeof timeSpecificity[number];
+}
+
+@customElement("md-timepicker")
+export class TimePicker extends LitElement {
+  @property({ type: Boolean }) twentyFourHourFormat = false;
+  @property({ type: Boolean }) editableHourOnly = false;
+  @property({ type: String }) timeSpecificity: TimePicker.TimeSpecificity = TIME_UNIT.SECOND;
+  @property({ type: String }) defaultTime = "12:00:00 AM";
+
```

Comment:
I think it'd be great to have a reflected attribute so that the selected time is live and visible, like a `value` or `selected-time` that shows the current selected time. Actually I don't see how the selected time is exposed for consumption at all, either as attribute or readable property.